dcap/types/src/certification_data.rs

```diff
@@ -44,13 +76,20 @@ impl<'a> TryFrom<&'a [u8]> for CertificationData<'a> {
 
         required += data_size;
         if actual < required {
-            Err(Quote3Error::InputLength { required, actual })
-        } else {
-            Ok(Self {
-                data: &bytes[..data_size],
-                data_type,
-            })
+            return Err(Quote3Error::InputLength { required, actual });
         }
+        let bytes = &bytes[..data_size];
+        let data = match data_type {
+            1 => CertificationData::Ppid(bytes),
+            2 => CertificationData::PpidEncryptedRsa2048(bytes),
+            3 => CertificationData::PpidEncryptedRsa3072(bytes),
+            4 => CertificationData::Pck(bytes),
+            5 => CertificationData::PckCertificateChain(PckCertificateChain { data: bytes }),
+            6 => CertificationData::EcdsaSignatureAuxData(bytes),
+            7 => CertificationData::PlatformManifest(bytes),
```

Comment:
I'd like to see the discriminant here as a constant, which may tend to lead to a slightly different design:

```rust
pub struct Ppid<'a>(&'a [u8]);

impl CertificationDataKind for Ppid {
    const KIND: u16 = 1;
}

pub enum CertificationData<'a> {
    Ppid(Ppid<'a>)
}
```

Reply:
Updated all variants to have unique types that wrap `&'a [u8]` and provide a `KIND` constant.